src/fileformats/ocd_georef_fields.cpp

```diff
@@ -0,0 +1,457 @@
+    { OcdGrid::Sweden, 2, 3006 }, // SWEREF99 TM
+    { OcdGrid::Sweden, 3, 3007 }, // SWEREF99 12 00
+    { OcdGrid::Sweden, 4, 3008 }, // SWEREF99 13 30
+    { OcdGrid::Sweden, 5, 3009 }, // SWEREF99 15 00
+    { OcdGrid::Sweden, 6, 3010 }, // SWEREF99 16 30
+    { OcdGrid::Sweden, 7, 3011 }, // SWEREF99 18 00
+    { OcdGrid::Sweden, 8, 3012 }, // SWEREF99 14 15
+    { OcdGrid::Sweden, 9, 3013 }, // SWEREF99 15 45
+    { OcdGrid::Sweden, 10, 3014 }, // SWEREF99 17 15
+    { OcdGrid::Sweden, 11, 3015 }, // SWEREF99 18 45
+    { OcdGrid::Sweden, 12, 3016 }, // SWEREF99 20 15
+    { OcdGrid::Sweden, 13, 3017 }, // SWEREF99 21 45
+    { OcdGrid::Sweden, 14, 3018 }, // SWEREF99 23 15
+    { OcdGrid::Switzerland, 1, 21781 }, // CH1903 / LV03 -- Swiss CH1903 / LV03
+    { OcdGrid::Switzerland, 2, 2056 }, // CH1903+ / LV95 -- Swiss CH1903+ / LV95
+};
+
+/**
+ * MapperCrsRecord struct is used for passing CRS information around within the
+ * translation code.
+ */
+struct MapperCrsRecord
+{
+	/** CRS id, converted to a string via crs_string_map and passed to
+	 *  CRSTemplateRegistry::find() */
+	const MapperCrs crs_id;
+	/** CRS parameter, converted to a string and passed through to
+	 *  CRSTemplateRegistry::specValues() */
+	const int crs_param;
+};
+
+/**
+ * Plain translation from OCD grid/zone combination into a Mapper
+ * scheme.
+ *
+ * @see translateMapperToOcd
+ */
+MapperCrsRecord translateOcdToMapper(const int ocd_grid_id,
+                                     const int ocd_zone_id)
+{
+	switch (static_cast<OcdGrid>(ocd_grid_id))
+	{
+	case OcdGrid::Local:
+		return { MapperCrs::Local, 0 };
+
+	case OcdGrid::Utm:
+		if (qAbs(ocd_zone_id) >= 1 && qAbs(ocd_zone_id) <= 60)
+			return { MapperCrs::Utm, ocd_zone_id };
+		break;
+
+	case OcdGrid::Germany:
+		// DHDN, Gauss-Krüger, Zone 2-5
+		if (ocd_zone_id >= 2 && ocd_zone_id <= 5)
+			return { MapperCrs::GaussKrueger, ocd_zone_id };
+		break;
+
+	default:
+		for (const auto& record : grid_zone_epsg_codes)
+		{
+			if (static_cast<int>(record.ocd_grid_id) == ocd_grid_id
+			    && record.ocd_zone_id == ocd_zone_id)
+				return { MapperCrs::Epsg, record.epsg_code };
+		}
+	}
+
+	return { MapperCrs::Invalid, 0 };
+}
+
+/**
+ * Imports string 1039 field i.
+ */
+void applyGridAndZone(Georeferencing& georef,
+                      const int combined_ocd_grid_zone,
+                      const std::function<void(const QString&)>& warning_handler)
+{
+	// while OCD codes can be negative, we are keeping grid_id always positive
+	// to match the enums; signum is carried in zone_id
+	const int ocd_grid_id = qAbs(combined_ocd_grid_zone / 1000),
+	          ocd_zone_id = qAbs(combined_ocd_grid_zone % 1000)
+	                        * (combined_ocd_grid_zone < 0 ? -1 : 1);
+
+	// Mapper has multiple ways of expressing some of the CRS's
+	// for now we are working with the first match
+	auto result = translateOcdToMapper(ocd_grid_id, ocd_zone_id);
```

Comment:
... this is the only time the split values are used.

Reply:
So be it. A sensible compiler will inline `translateOcdToMapper()` into `applyGridAndZone()` so there are no gains nor losses from the signature change. I'll also drop `OcdCrsRecord`, which is the same case.

Reply:
My argument (in the previous comment) was not a sensible compiler, but comprehensibility for humans. "Named constructor" is one point, and avoiding arguments of same type next to each other is another point. Cf.
- [C++ FAQ on the named constructor idiom](https://isocpp.org/wiki/faq/ctors#named-ctor-idiom)
- [C++ Core Guidelines:  Avoid adjacent unrelated parameters of the same type](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#Ri-unrelated)
- Recent blog entry linked on isocpp.org on [certain constructors in the STL which got it wrong](https://isocpp.org/blog/2018/03/strings-competing-constructors-andrzej-krzemieski).

Note that the named constructors don't interfere with the Rule of Zero.

Reply:
I'm aware of the recommendations and acknowledge that this parameter arrangement is against recommendations stated in Guidelines. I question whether fixing parameter count is worth the effort associated with rework of the patch set. Anyway, the other points warrant the respin so this one is included.